datajunction-server/datajunction_server/internal/nodes.py

```diff
@@ -534,13 +533,21 @@ def update_node_with_query(
                     ),
                 ),
             )
-        schedule_materialization_jobs(
-            new_revision.materializations,
-            query_service_client,  # type: ignore
+        background_tasks.add_task(
+            schedule_materialization_jobs,
+            materializations=node.current.materializations,
+            query_service_client=query_service_client,
         )
         session.add(new_revision)
         session.commit()
 
+    if background_tasks:
```

Comment:
Hmm. So why you didn't check the existence of background_tasks when adding `schedule_materialization_jobs` to it?

Reply:
So that's because we always pass in that argument, but I think I'll include that check anyway just in case this changes in the future